# origin: CAN-2004-0394
# cset: 1.1458.3.10
# inclusion: n/a (projected: 2.4.28-pre3)
# description: fix possible buffer overflow in panic()
# revision date: 2004-09-05

# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/08/12 18:45:04-03:00 chrisw@osdl.org 
#   [PATCH] fix possible buffer overflow in panic()
#   
#   Minor issue here with the static buffer in 2.4 kernel's panic() function.
#   This has already been fixed in 2.6.
#   
#   [PATCH] fix possible buffer overflow in panic()
#   
#   ===== kernel/panic.c 1.12 vs edited =====
# 
# kernel/panic.c
#   2004/04/15 07:33:32-03:00 chrisw@osdl.org +1 -1
#   fix possible buffer overflow in panic()
# 
diff -Nru a/kernel/panic.c b/kernel/panic.c
--- a/kernel/panic.c	2004-09-05 02:07:39 -07:00
+++ b/kernel/panic.c	2004-09-05 02:07:39 -07:00
@@ -59,7 +59,7 @@
 	
 	bust_spinlocks(1);
 	va_start(args, fmt);
-	vsprintf(buf, fmt, args);
+	vsnprintf(buf, sizeof(buf), fmt, args);
 	va_end(args);
 	printk(KERN_EMERG "Kernel panic: %s\n",buf);
 	if (in_interrupt())
